### PR TITLE
Issue/2497 google analytics

### PR DESCRIPTION
**Pull request by mkeilman**
Not sure I got this right but want the PR in place since #2497 is high priority

### CODE REVIEW

File: sirepo/server.py [hunk 1]
```diff
@@ -704,5 +711,6 @@ def static_dir(dir_name):
 cfg = pkconfig.init(
     enable_source_cache_key=(True, bool, 'enable source cache key, disable to allow local file edits in Chrome'),
     db_dir=pkconfig.ReplacedBy('sirepo.srdb.root'),
+    google_tag_manager_id=(os.environ.get('SIREPO_SERVER_GOOGLE_TAG_MANAGER_ID', None), str, 'tag mgr id from google'),
```
Comment (robnagler) [inline sirepo/server.py:714]: pkconfig reads os.environ so just:
```py
    google_tag_manager_id=(None, str, 'enable google analytics with this id'),
```

File: tests/server_test.py [hunk 2]
```diff
@@ -80,13 +80,23 @@ def test_elegant_data_file(fc):
...
     r = fc.get('/old')
     assert 'LandingPageController' in pkcompat.from_bytes(r.data), \
         'Top level document is the landing page'
     r = fc.get('/robots.txt')
     pkunit.pkre('elegant.*myapp.*srw', pkcompat.from_bytes(r.data))
 
+    r = fc.get('/en/landing.html')
+    pkok(
+        not re.search(
+            r'https://www.googletagmanager.com/gtm.js\?id=',
```
Comment (robnagler) [inline tests/server_test.py:94]: The looser the test the more maintainable, e.g. `re.search('googletag'`.

File: tests/server_test.py [hunk 3]
```diff
@@ -80,13 +80,23 @@ def test_elegant_data_file(fc):
...
+            pkcompat.from_bytes(r.data)
+        ),
+        'Unexpected injection of google tag mgr',
```
Comment (robnagler) [inline tests/server_test.py:97]: dump the data for debugging: `'Unexpected injection of googletag data={}', r.data`

File: tests/google_tag_manager_test.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    from pykern import pkcompat, pkunit
+    from pykern.pkdebug import pkdc, pkdp, pkdlog
+    from pykern.pkunit import pkeq, pkok, pkre
+    import re
+
+    # test non-static page
+    r = fc.get('myapp')
+    pkok(
+        not re.search(
+            r'https://www.googletagmanager.com/gtm.js\?id=',
```
Comment (robnagler) [inline tests/google_tag_manager_test.py:30]: Same as below.

File: tests/google_tag_manager_test.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            pkcompat.from_bytes(r.data)
+        ),
+        'Unexpected injection of google tag mgr'
+    )
+
+    # test successful injection
+    r = fc.get('/en/landing.html')
+    pkok(
+        re.search(_NO_SUCH_ID, pkcompat.from_bytes(r.data)),
```
Comment (robnagler) [inline tests/google_tag_manager_test.py:39]: `_NO_SUCH_ID` is `_EXPECT_ID`. We are expecting to see this id. "no such" implies we aren't expecting it.

File: tests/google_tag_manager_test.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        not re.search(
+            r'https://www.googletagmanager.com/gtm.js\?id=',
+            pkcompat.from_bytes(r.data)
+        ),
+        'Unexpected injection of google tag mgr'
+    )
+
+    # test successful injection
+    r = fc.get('/en/landing.html')
+    pkok(
```
Comment (robnagler) [inline tests/google_tag_manager_test.py:38]: `pkre` does the re.search so simpler to use.

File: sirepo/server.py [hunk 7]
```diff
@@ -519,6 +521,11 @@ def api_staticFile(path_info=None):
     Returns:
         flask.Response: flask.send_from_directory response
     """
+    if re.match(r'en/.*html', path_info):
```
Comment (robnagler) [inline sirepo/server.py:524]: `path_info` can be None, so protect against this. 

I changed my mind. The analytics code is copied N times so what we want to do is something like this:

```py
if _google_tag_manager and re.match(r'en/.*html', path_info or ''):
    return _google_tag_manager_re.sub(
        _google_tag_manager, 
        pkio.read_text(simulation_db.STATIC_FOLDER.join(path_info)),
    )
```

At the top of the file:

```py
#: If google_tag_manager_id set, string to insert in landing pages for google analytics
_google_tag_manager = None

#: what to match in landing pages to insert `_google_tag_manager`
_google_tag_manager_re = re.compile('(?=</head>)', flags=re.IGNORECASE)
```

Then add to`init()`:

```py
    if cfg.google_tag_manager_id:
        _google_tag_manager = f'''<script>
(function(w,d,s,l,i){{w[l]=w[l]||[];w[l].push({{'gtm.start':new Date().getTime(),event:'gtm.js'}});var f=d.getElementsByTagName(s)[0],j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src='https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);}})(window,document,'script','dataLayer','{cfg.google_tag_manager_id}');
</script>'''
```

File: sirepo/package_data/static/en/landing.html.jinja [hunk 8]
```diff
@@ -21,6 +21,11 @@
   <script src="https://stackpath.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
   <script src="https://cdnjs.cloudflare.com/ajax/libs/showdown/1.9.0/showdown.min.js"></script>
   <script src="/en/js/landing.js"></script>
+{% if google_tag_manager_id %}
```
Comment (robnagler) [inline sirepo/package_data/static/en/landing.html.jinja:24]: remove this and rename back to .html

**Comment by mkeilman** [comment]
Ready for 2nd pass

**Pull request merged**